akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -510,6 +518,20 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  //this limits number of messages that can be processed synchronously during one actor receive.
+  private val shortCircuitLimit: Int = _initial.mat.settings.syncProcessingLimit
+  private var shortCircuitBuffer: ByteBuffer[BoundaryEvent] = _
+  private var currentLimit: Int = shortCircuitLimit
```

Comment:
>How does syncProcessingLimit relate to eventLimit?

eventLimit is about processing connection events and maximum for this limit is "number of connections" * 2. This limit works within one shell.
syncProcessingLimit works across several shells for one actor and mazimum should relate to number of shells in this actor. The reason have it here is that from some cases there is a ping pong (asyncCalls) between stream and substream. So actor receive method can process (synchronously) such events very long time. 

Still idea is very similar to for eventLimit and syncProcessingLimit. We are doing sync messages processing for asyncCalls only for for now (which is futures and stream to substream communications). Potentially we can add other events to shortCircuit queue to get rid of actor async overhead.